roles/linux/opensearch/tasks/security.yml

```diff
@@ -137,21 +225,59 @@
   environment:
     JAVA_HOME: "{{ os_home }}/jdk"
   run_once: true
-  when: configuration.changed
+  when: configuration.changed or iac_enable
+
+- name: Security Plugin configuration | Check that the files/internal_users.yml exists
+  stat:
+    path: files/internal_users.yml
+  register: custom_users_result
+  delegate_to: localhost
+  run_once: true
+
+- name: Security Plugin configuration | Check for a custom configuration for internal users and hash passwords for them
+  block:
+
+    - name: Security Plugin configuration | Load custom internal users configuration
+      include_vars:
+        file: files/internal_users.yml
+        name: custom_users
+      run_once: true
+
+    - name: Security Plugin configuration | Filter service keys from the list of users
+      set_fact:
+        custom_users_filtered: '{{ custom_users|dict2items|rejectattr("key", "equalto", "_meta")|list|items2dict }}'
+
+#    - name: Security Plugin configuration | debug
+#      debug:
+#        msg: "{{ item }} password: {{ lookup('vars', item + '_password') }}"
+#      with_items: "{{ custom_users_filtered }}"
+#      run_once: true
+
+    - name: Security Plugin configuration | Set passwords for all users from custom config
+      shell: >
+        sed -i '/hash: / s,{{ lookup('vars', item + '_password') }},'$(bash {{ os_sec_plugin_tools_path }}/hash.sh -p {{ lookup('vars', item + '_password') }} | tail -1)','
+        {{ os_sec_plugin_conf_path }}/internal_users.yml
```

Comment:
In principle, this is possible. But we need to redo the logic of working with user passwords. This task is done by analogy with the task "- name: Security Plugin configuration | Set the Admin user password". Now passwords are transmitted in clear text and hashed when performing the role. If you transmit hashes, this will lead to the need for additional actions on the part of users (pre-downloading the container from opensearch locally, hashing each password), which is not very convenient, I think.